src/main/java/fr/jcgay/maven/extension/revision/UniqueRevisionFiltering.java

```diff
@@ -45,13 +45,13 @@ public Artifact transformArtifact(Artifact artifact) {
         boolean hasRevisionNumber = false;
         if (isRevision(pom.getVersion())) {
             hasRevisionNumber = true;
-            pom.setVersion(artifact.getBaseVersion());
+            pom.setVersion(resolve(pom.getVersion(), artifact.getBaseVersion()));
```

Comment:
I think you don't need to change the code here, `artifact.getBaseVersion()` is already the final version we want resolved by Maven is't it ?